bpf/tests/prog_test/prog_test.go

```diff
@@ -341,6 +341,63 @@ func testCt4Rst(spec *ebpf.Collection) error {
 	return nil
 }
 
+func testCT4ICMPFragNeeded(spec *ebpf.Collection) error {
+	prog := spec.Programs["test_ct4_icmp_frag_needed"]
+	if prog == nil {
+		return errors.New("did not find test_ct4_icmp_frag_needed program")
+	}
+
+	packetOrig := gopacket.NewSerializeBuffer()
+	gopacket.SerializeLayers(packetOrig, gopacket.SerializeOptions{},
+		&layers.IPv4{
+			Version:  4,
+			SrcIP:    net.IP{1, 1, 1, 1},
+			DstIP:    net.IP{2, 2, 2, 1},
+			Protocol: layers.IPProtocolICMPv4,
+			IHL:      5,
+		},
+		&layers.ICMPv4{
+			TypeCode: layers.CreateICMPv4TypeCode(
+				layers.ICMPv4TypeEchoRequest,
+				0),
+			Id: 123,
+		})
+
+	packetIn := gopacket.NewSerializeBuffer()
+	gopacket.SerializeLayers(packetIn, gopacket.SerializeOptions{},
+		&layers.Ethernet{
+			SrcMAC:       net.HardwareAddr{1, 0, 0, 3, 0, 10},
+			DstMAC:       net.HardwareAddr{1, 0, 0, 3, 0, 20},
+			EthernetType: layers.EthernetTypeIPv4,
+		},
+		&layers.IPv4{
+			Version: 4,
+			// The router is responding with an ICMP need to Frag.
+			SrcIP:    net.IP{0, 0, 0, 0},
+			DstIP:    net.IP{3, 3, 3, 1},
+			Protocol: layers.IPProtocolICMPv4,
+			IHL:      5,
+		},
+		&layers.ICMPv4{
+			TypeCode: layers.CreateICMPv4TypeCode(
+				layers.ICMPv4TypeDestinationUnreachable,
+				layers.ICMPv4CodeFragmentationNeeded),
+		},
+		gopacket.Payload(packetOrig.Bytes()),
+	)
+
+	bpfRet, _, err := prog.Test(packetIn.Bytes())
+	if err != nil {
```

Comment:
I believe the second argument that is omitted here is the output packet?
Are there any specific expectations that we have for the output packet? would it make sense to test them?

Reply:
I didn't think about testing that one I was more focus on ensuring behavior of the NAT process. I will have a look and let you know.